elytron/src/main/java/org/wildfly/extension/elytron/KeyStoreDefinition.java

```diff
@@ -82,9 +82,10 @@
 
     static final ServiceUtil<KeyStore> KEY_STORE_UTIL = ServiceUtil.newInstance(KEY_STORE_RUNTIME_CAPABILITY, ElytronDescriptionConstants.KEY_STORE, KeyStore.class);
 
-    static final SimpleAttributeDefinition TYPE = new SimpleAttributeDefinitionBuilder(ElytronDescriptionConstants.TYPE, ModelType.STRING, false)
+    static final SimpleAttributeDefinition TYPE = new SimpleAttributeDefinitionBuilder(ElytronDescriptionConstants.TYPE, ModelType.STRING, true)
         .setAttributeGroup(ElytronDescriptionConstants.IMPLEMENTATION)
         .setAllowExpression(true)
+        .setDefaultValue(new ModelNode(ElytronDescriptionConstants.AUTOMATIC))
```

Comment:
From discussion on proposal, I thought there will be no special type introduced. [1]

[1] https://github.com/wildfly/wildfly-proposals/pull/124#discussion_r212284418

Reply:
+1 A main motivator for this is to reduce the configuration required so the attribute should just become optional - if the attribute is undefined that becomes our trigger to start automatically detecting it.

Reply:
Sorry for that one, fixed